src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -568,14 +568,18 @@ matches deps TypeClassDictionaryInScope{..} tys =
 -- | Add a dictionary for the constraint to the scope, and dictionaries
 -- for all implied superclass instances.
 newDictionaries
-  :: MonadState CheckState m
+  :: (MonadState CheckState m, MonadError MultipleErrors m)
   => [(Qualified (ProperName 'ClassName), Integer)]
   -> Qualified Ident
   -> Constraint
   -> m [NamedDict]
 newDictionaries path name (Constraint className instanceTy _) = do
     tcs <- gets (typeClasses . checkEnv)
-    let TypeClassData{..} = fromMaybe (internalError "newDictionaries: type class lookup failed") $ M.lookup className tcs
+    TypeClassData{..} <-
+      maybe
+        (throwError . errorMessage . UnknownClass $ className)
+        pure
+        $ M.lookup className tcs
```

Comment:
If there's a more suitable idiom, I'd be happy to give it a shot :slightly_smiling_face: